src/main/java/org/eclipse/tractusx/managedidentitywallets/domain/command/CredentialSearchCommand.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * *******************************************************************************
+ *  Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations
+ *  under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ * ******************************************************************************
+ */
+
+package org.eclipse.tractusx.managedidentitywallets.domain.command;
+
+import com.smartsensesolutions.java.commons.sort.Sort;
+import com.smartsensesolutions.java.commons.sort.SortType;
+import org.eclipse.tractusx.managedidentitywallets.domain.BPN;
+import org.eclipse.tractusx.managedidentitywallets.domain.CredentialId;
+import org.eclipse.tractusx.managedidentitywallets.domain.Identifier;
+import org.eclipse.tractusx.managedidentitywallets.domain.SortColumn;
+import org.eclipse.tractusx.managedidentitywallets.domain.TypeToSearch;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * CredentialSearch is a search for credentials.
+ */
+public class CredentialSearchCommand {
+
+    private final CredentialId credentialId;
+
+    private final Identifier identifier;
+
+    private final List<TypeToSearch> typesToSearch;
+
+    private final Sort sort;
+
+    private final int pageNumber;
+
+    private final int pageSize;
+
+    private final BPN callerBpn;
+
+    private CredentialSearchCommand(Builder builder) {
+        credentialId = builder.id;
+        identifier = builder.identifier;
+        typesToSearch = builder.typesToSearch;
+        sort = builder.sort;
+        pageNumber = builder.pageNumber;
+        pageSize = builder.pageSize;
+        callerBpn = builder.callerBpn;
+    }
+
+    public CredentialId credentialId() {
+        return credentialId;
+    }
+
+    public Identifier identifier() {
```

Comment:
similar recommendation for differenciation between DID and identifier, here it about DID